app/controllers/comments_controller.rb

```diff
@@ -1,10 +1,39 @@
 class CommentsController < ApplicationController
+  def new
+    @article = Article.find(params[:id])
+    @comment = @article.comments.new
+  end
+
   def create
     @article = Article.find(params[:article_id])
     @comment = @article.comments.create(comment_params)
     redirect_to article_path(@article)
   end
 
+  def edit
+    @article = Article.find(params[:article_id])
```

Comment:
move this to private and call it from before_action

Reply:
there are 2 ways to do this
- first one just like toon said call with `before_action`
- another one is something like this below and call using only `article` (this way like request cache)
ref: https://stackoverflow.com/questions/3602518/how-do-i-cache-a-method-with-ruby-rails
```
def article
  @article ||= Article.find(params[:article_id])
end
```